python/salpytools/states.py

```diff
@@ -163,5 +163,104 @@ def wake(self):
         self.log.debug('State waking')
         self.send_logEvent('SummaryState')
 
+    #<----- Default State methods corresponding to UML design under here ------>
+
+    def disable(self, model):
+        print("Cannot transition to Disbale state")
```

Comment:
Shouldn't this be an exception or something like that? 

We can either throw an exception when there's an error or return an error code... On Python, I prefer the former (throw exception)... 

I also think we should create an exception specifically for when there's a "cannot transition to state". I would suggest something like;

```
class StateTransitionException(Exception):
    pass
```

Also, need to be done on all of them.

Reply:
After looking into it more it seems that a popular design is to not throw exceptions but go into a fault state with some error handling, https://softwareengineering.stackexchange.com/questions/349998/how-should-state-machines-handle-exceptions-in-actions. Is there any validity to this? 

Anyhow I am throwing suggestions like you suggested.